backend/src/openbeheer/catalogi/api/views.py

```diff
@@ -78,9 +78,15 @@ def get(self, request: Request, slug: str) -> Response:
                     if catalogue.naam
                     else catalogue.domein
                 )
-                # OZ API specs say that is is not required, but VNG specs say it is.
+                # OZ API specs say that url is not required, but VNG specs say it is.
                 # In practice, it is always present.
-                value = catalogue.url
+                url = catalogue.url
+
+                # Extract the UUID by removing service API root and known endpoint.
+                # We only support 1 ZTC for now.
+                # TODO: Provide util for this?
+                service = Service.objects.get(api_type=APITypes.ztc)
+                value = url.removeprefix(service.api_root + "catalogussen/")
```

Comment:
That util is called ztc_client()... While you're at it, pass `slug` to it and remove that comment.

No need to query the database inside this loop :grimacing: 

```suggestion
                value = url.removeprefix(f"{client.api_root}catalogussen/")
```

But seriously, we might want a util to handle these mappings. But not right here, right now. Let's see more cases first.